tripy/docs/post0_developer_guides/how-to-add-new-ops.md

```diff
@@ -192,9 +191,12 @@ from tripy.types import ShapeLike
 # `autodoc_options` parameter.
 @export.public_api(document_under="tensor_operations")
 
-# The `convert_to_tensors` decorator automatically converts compatible
-# arguments, like `TensorLike` or `ShapeLike`s, into tensors.
-@frontend_utils.convert_to_tensors()
+# We can use the `constraints.interface` decorator to specify constraints on
+# inputs and perform transformations on them, like automatically converting
+# compatible arguments (e.g., `TensorLike` or `ShapeLike`s) into tensors.
+# We will aim to include most constraints and transformations in this decorator
+# so as to avoid layering too many decorators.
+@constraints.interface(convert_tensor_and_shape_likes=True)
```

Comment:
I think `convert_to_tensors` is a more accurate reflection of what we're doing. The types that we support (i.e. `TensorLike` and `ShapeLike`) are just a detail that might change over time. OTOH the behavior (i.e. converting things to `Tensor`s) should remain consistent.